### PR TITLE
Track errors from SCMStatusReporter in WorkflowRuns

### PR DESCRIPTION
**Pull request by krauselukas**
In the past we only wrote the exception raised in the scm status reporter to the logfiles, which are not accessible for the users. In order to help the users to debug, we want to show the errors in the workflow runs UI. For now simply by writing it to the response body. 

*How to test:*

Easiest way to test this is by simply setting up a workflow with a `branch_package` step and to mock the code by raising one of the `Octokit` exceptions if the `SCMStatusReporter#call` or somehow let the reporter fail by for example cutting network connection... 

*TODO:*

- [x] Add testcase to `scm_status_reporter_spec.rb` to check that workflow_run tracks error
- [x] Make sure migration can be run without downtime, adjust when necessary @krauselukas 
~- [ ] "Translate" exception message from Octokit to something meaningful~ (follow up PR)

### CODE REVIEW

**Comment by krauselukas** [comment]
[Track errors from SCMStatusReporter in WorkflowRuns (M)](https://trello.com/c/V6E2JPig/1819-track-errors-from-scmstatusreporter-in-workflowruns-m)

**Comment by rubhanazeem** [comment]
@krauselukas some related specs are failing

**Comment by krauselukas** [comment]
> @krauselukas some related specs are failing

Fixed it :)

File: src/api/app/models/token/workflow.rb [hunk 1]
```diff
@@ -20,9 +19,9 @@ def call(options)
     return validation_errors unless validation_errors.none?
 
     # This is just an initial generic report to give a feedback asap. Initial status pending
-    ScmInitialStatusReporter.new(@scm_webhook.payload, @scm_webhook.payload, scm_token, workflow_run).call
+    ScmInitialStatusReporter.new(@scm_webhook.payload, @scm_webhook.payload, scm_token, workflow_run.reload).call
```
Comment (hennevogel) [inline src/api/app/models/token/workflow.rb:22]: Why reload?

File: src/api/app/services/scm_exception_handler.rb [hunk 2]
```diff
@@ -36,20 +35,19 @@ class SCMExceptionHandler
...
+    @workflow_run = workflow_run
   end
 
   private
 
-  def log(exception)
-    token = Token::Workflow.find_by(scm_token: @scm_token)
-    Rails.logger.error "#{exception.class}: #{exception.message}. TokenID: #{token.id}, User: #{token.user.login}, Event Subscription Payload: #{@event_subscription_payload}"
+  def log_to_workflow_run(exception)
+    @workflow_run.update_to_fail("Failed to report back to SCM: #{exception.message}")
```
Comment (hennevogel) [inline src/api/app/services/scm_exception_handler.rb:51]: You know which SCM right?

File: src/api/app/services/scm_initial_status_reporter.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class ScmInitialStatusReporter < SCMStatusReporter
   attr_accessor :state
 
   def initialize(event_payload, event_subscription_payload, scm_token, workflow_run, event_type = nil)
-    super(event_payload, event_subscription_payload, scm_token)
+    super(event_payload, event_subscription_payload, scm_token, nil, workflow_run.id)
```
Comment (hennevogel) [inline src/api/app/services/scm_initial_status_reporter.rb:5]: Hm why pass down `nil` if it's never not `nil`?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/app/services/scm_initial_status_reporter.rb (hunk 3 above)
Comment (rubhanazeem) [inline src/api/app/services/scm_initial_status_reporter.rb:5]: Shouldn't it be?
```
super(event_payload, event_subscription_payload, scm_token, event_type, workflow_run.id)
```

Comment (krauselukas) [reply src/api/app/services/scm_initial_status_reporter.rb:5]: true

File: src/api/app/models/token/workflow.rb (hunk 1 above)
Comment (krauselukas) [reply src/api/app/models/token/workflow.rb]: used it at some point, no reason anymore, I removed it.

File: src/api/app/services/scm_exception_handler.rb (hunk 2 above)
Comment (krauselukas) [reply src/api/app/services/scm_exception_handler.rb]: jupp :)

File: src/api/app/services/scm_initial_status_reporter.rb (hunk 3 above)
Comment (krauselukas) [reply src/api/app/services/scm_initial_status_reporter.rb]: :+1:

**Comment by krauselukas** [comment]
@hennevogel @rubhanazeem please have another look :)

**Review by rubhanazeem — APPROVED** [review]

**Pull request merged**